metrics/internal/collectors/ceph-block-pool.go

```diff
@@ -0,0 +1,154 @@
+package collectors
+
+import (
+	"github.com/openshift/ocs-operator/metrics/internal/options"
+	"github.com/prometheus/client_golang/prometheus"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	cephv1listers "github.com/rook/rook/pkg/client/listers/ceph.rook.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+const (
+	// component within the project/exporter
+	rbdSubsystem = "rbd_mirroring"
+)
+
+var _ prometheus.Collector = &CephBlockPoolCollector{}
+
+// CephRBDMirroringCollector is a custom collector for CephBlockPool Custom Resource
+type CephBlockPoolCollector struct {
+	MirroringImageHealth *prometheus.Desc
+	MirroringStatus      *prometheus.Desc
+	Informer             cache.SharedIndexInformer
+	AllowedNamespaces    []string
+}
+
+// NewCephRBDMirroringCollector constructs a collector
+func NewCephBlockPoolCollector(opts *options.Options) *CephBlockPoolCollector {
+	client, err := rookclient.NewForConfig(opts.Kubeconfig)
+	if err != nil {
+		klog.Error(err)
+	}
+
+	lw := cache.NewListWatchFromClient(client.CephV1().RESTClient(), "cephblockpools", metav1.NamespaceAll, fields.Everything())
+	sharedIndexInformer := cache.NewSharedIndexInformer(lw, &cephv1.CephBlockPool{}, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+
+	return &CephBlockPoolCollector{
+		MirroringImageHealth: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, rbdSubsystem, "image_health"),
+			`RBD Mirroring Image Health. 0=OK, 1=WARNING & 2=ERROR`,
+			[]string{"name", "namespace"},
+			nil,
+		),
+		MirroringStatus: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, rbdSubsystem, "status"),
+			`RBD Mirroring Image Health. 0=Enabled, 1=Disabled`,
+			[]string{"name", "namespace"},
+			nil,
```

Comment:
I think we should add another label for the storage cluster name. When there are multiple storage systems in the Clusters, that could be used to differentiate the metrics.